### ISSUE TITLE
torch.jit.script gives false results for autograd if complex data types are involved

### ISSUE THREAD

**Issue by frank-roesler** [pytorch/pytorch#154324]
### 🐛 Describe the bug

If the decorator @torch.jit.script is used to speed up a function that depends on complex tensors, then later invocation of the .backward() method seems to only compute the real parts of the gradients, but ignore the imaginary parts.
Here is a minimal working example:

```
import torch

# @torch.jit.script
def f(x: torch.Tensor) -> torch.Tensor:
    x = x * torch.conj(x) + 1.0
    return x.real

lr = 1e-1
steps = 100

x = torch.randn(10, dtype=torch.complex64, requires_grad=True)

for step in range(steps):
    y = f(x)
    loss = torch.mean(y)
    losses.append(loss.item())
    loss.backward()
    with torch.no_grad():
        x -= lr * x.grad
        x.grad.zero_()
```
Without the line "@torch.jit.script" the minimization works fine and x converges to 0 (as it should, because |x|**2 gets minimized). If "@torch.jit.script" is added, the loss explodes and only the real part of x converges to 0, while the imaginary part grows.

If I print x.grad, I get (after 50 steps using manual seed): 
* without jit:  
x.grad.real = 6.7958694671688136e-06
x.grad.imag = 2.599932031444041e-06
* with jit:  
x.grad.real = 6.7958694671688136e-06
x.grad.imag = -1105.181640625

Expected behaviour:  
The gradients with and without torch.jit.script should be equal and the minimization should yield the same results.

### Versions
PyTorch version: 2.7.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.5 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.0.13.3)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.22 (main, Apr  8 2025, 15:21:55)  [Clang 16.0.0 (clang-1600.0.26.6)] (64-bit runtime)
Python platform: macOS-15.5-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1

Versions of relevant libraries:
[pip3] numpy==2.0.2
[pip3] torch==2.7.0

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel @ezyang @anjali411 @dylanbespalko @mruberry @nikitaved @amjames

**Comment by malfet** [comment]
Thank you for reporting the issue. Please note that TorchScript is in maintenance mode right now.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
